### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by PhenomRetroShare**

### CODE REVIEW

File: libretroshare/src/util/rsdir.cc [hunk 1]
```diff
@@ -327,14 +327,14 @@ bool RsDirUtil::copyFile(const std::string& source,const std::string& dest)
 #else
 	FILE *in = fopen64(source.c_str(),"rb") ;
 
-	if(in == NULL)
+	if(in == nullptr)
```
Comment (G10h4ck) [inline libretroshare/src/util/rsdir.cc]: This and similar should be replaced by `if(!in)`

**Review by G10h4ck — CHANGES_REQUESTED** [review]

File: libretroshare/src/util/rsdir.cc (hunk 1 above)
Comment (PhenomRetroShare) [reply libretroshare/src/util/rsdir.cc]: Done. Will be more concern now. ;)

File: libretroshare/src/ft/ftserver.cc [hunk 2]
```diff
@@ -223,18 +223,18 @@ void ftServer::StopThreads()
 	mFtExtra->join();
 
 	delete (mFtDataplex);
```
Comment (G10h4ck) [inline libretroshare/src/ft/ftserver.cc:225]: WTF are those?? those pointers are just nullptr as it is visible on the initializer list, those couple of lines should disappear and followinf similars too

**Review by G10h4ck — CHANGES_REQUESTED** [review]

File: libretroshare/src/turtle/p3turtle.cc [hunk 3]
```diff
@@ -92,11 +88,11 @@ static const uint32_t TURTLE_SEARCH_RESULT_MAX_HITS_DEFAULT= 100 ;		/// default
 
 static const float depth_peer_probability[7] = { 1.0f,0.99f,0.9f,0.7f,0.6f,0.5,0.4f } ;
 
-static const int TUNNEL_REQUEST_PACKET_SIZE             = 50 ;
+static const float TUNNEL_REQUEST_PACKET_SIZE           = 50.0f ;
```
Comment (G10h4ck) [inline libretroshare/src/turtle/p3turtle.cc]: This doesn't seems like it should be a float, maybe the error is elsewhere where this constant is used

**Review by G10h4ck — CHANGES_REQUESTED** [review]

File: libretroshare/src/ft/ftserver.cc (hunk 2 above)
Comment (csoler) [reply libretroshare/src/ft/ftserver.cc:225]: I'd be surprised these pointers are not set anywhere ;-) without them, there's no FT (especially mFtDataplex).

File: libretroshare/src/turtle/p3turtle.cc (hunk 3 above)
Comment (csoler) [reply libretroshare/src/turtle/p3turtle.cc]: agreed: we may compute some float value elsewhere that is compared to this const. Would be better to cast the other thing to int, or the const to float only when it is used.

File: libretroshare/src/ft/ftserver.cc (hunk 2 above)
Comment (PhenomRetroShare) [reply libretroshare/src/ft/ftserver.cc:225]: They are initialized in 133:
void ftServer::SetupFtServer()
and used on all code...
I don't understand your comment @G10h4ck.

File: libretroshare/src/turtle/p3turtle.cc (hunk 3 above)
Comment (PhenomRetroShare) [reply libretroshare/src/turtle/p3turtle.cc]: This const is used only with float values.
So why cast it to float and not defined it to float at start?

File: libretroshare/src/ft/ftserver.cc (hunk 2 above)
Comment (G10h4ck) [reply libretroshare/src/ft/ftserver.cc:225]: Sorry Github diff view fooled me those deletes seemed to be inside the constructor hence my WTF reaction

File: libretroshare/src/turtle/p3turtle.cc (hunk 3 above)
Comment (G10h4ck) [reply libretroshare/src/turtle/p3turtle.cc]: Then maybe the name of the constant is misleading? Calling it `*_SIZE` one expect it to be size in bytes or similar thing

Comment (csoler) [reply libretroshare/src/turtle/p3turtle.cc]: it should be a number of bytes. Phenom's option is best: just cast it to float during comparison.

Comment (PhenomRetroShare) [reply libretroshare/src/turtle/p3turtle.cc]: Perhaps I've explained badly.
Cast was done before (old style) during comparison. But as this value only used with float operators, why don't define it as float?
Look at [libretroshare/src/turtle/p3turtle.cc:1749](https://github.com/RetroShare/RetroShare/blob/master/libretroshare/src/turtle/p3turtle.cc#L1749)

Comment (G10h4ck) [reply libretroshare/src/turtle/p3turtle.cc]: In this case I would rename it as `TUNNEL_REQUEST_PACKET_SIZE_APPROX` and keep it as float

Comment (PhenomRetroShare) [reply libretroshare/src/turtle/p3turtle.cc]: Done.

File: libretroshare/src/ft/ftextralist.cc [hunk 4]
```diff
@@ -111,7 +111,7 @@ void ftExtraList::hashAFile()
 #endif
 
 	/* hash it! */
-	std::string name, hash;
+	//std::string name, hash;
```
Comment (G10h4ck) [inline libretroshare/src/ft/ftextralist.cc]: better remove dead-code entirely ;)

File: libretroshare/src/ft/ftextralist.cc [hunk 5]
```diff
@@ -466,16 +466,16 @@ bool    ftExtraList::loadList(std::list<RsItem *>& load)
 
 		/* open file */
 		FILE *fd = RsDirUtil::rs_fopen(fi->file.path.c_str(), "rb");
-		if (fd == NULL)
+		if (fd == nullptr)
```
Comment (G10h4ck) [inline libretroshare/src/ft/ftextralist.cc]: better `if(!fd)`

File: libretroshare/src/ft/ftserver.cc [hunk 6]
```diff
@@ -1356,7 +1357,7 @@ bool ftServer::decryptItem(const RsTurtleGenericDataItem *encrypted_item,const R
 	decrypted_item = dynamic_cast<RsTurtleGenericTunnelItem*>(deserialise(data,&data_size)) ;
 	free(data);
 
-	return (decrypted_item != NULL);
+	return (decrypted_item);
```
Comment (G10h4ck) [inline libretroshare/src/ft/ftserver.cc]: Are the parentheses needed?

File: libretroshare/src/ft/ftserver.cc [hunk 7]
```diff
@@ -1556,7 +1559,7 @@ void ftServer::receiveTurtleData(const RsTurtleGenericTunnelItem *i,
 #endif
 			getMultiplexer()->recvData(virtual_peer_id,hash,0,item->chunk_offset,item->chunk_size,item->chunk_data) ;
 
-			const_cast<RsTurtleFileDataItem*>(item)->chunk_data = NULL ;	// this prevents deletion in the destructor of RsFileDataItem, because data will be deleted
```
Comment (G10h4ck) [inline libretroshare/src/ft/ftserver.cc:1559]: @csoler Shouldn't the method signature changed removing `const` instead of using the canst in this case? Or it causes a refactor chain reaction?

File: libretroshare/src/gxs/rsgxsnetservice.cc [hunk 8]
```diff
@@ -3801,7 +3809,7 @@ bool RsGxsNetService::processTransactionForDecryption(NxsTransaction *tr)
     GXSNETDEBUG_P_(peerId) << "RsGxsNetService::decryptTransaction()" << std::endl;
 #endif
 
-    std::list<RsNxsItem*> decrypted_items ;
+	//std::list<RsNxsItem*> decrypted_items ;
```
Comment (G10h4ck) [inline libretroshare/src/gxs/rsgxsnetservice.cc]: Better removing dead-code completely

File: libretroshare/src/gxs/rsgxsnetservice.cc [hunk 9]
```diff
@@ -3810,7 +3818,7 @@ bool RsGxsNetService::processTransactionForDecryption(NxsTransaction *tr)
     {
         RsNxsEncryptedDataItem *encrypted_item = dynamic_cast<RsNxsEncryptedDataItem*>(*it) ;
 
-        if(encrypted_item == NULL)
+        if(encrypted_item == nullptr)
```
Comment (G10h4ck) [inline libretroshare/src/gxs/rsgxsnetservice.cc]: better `if(!encrypted_item)`

File: libretroshare/src/gxs/rsgxsutil.cc [hunk 10]
```diff
@@ -147,8 +147,8 @@ bool RsGxsMessageCleanUp::clean()
 
 RsGxsIntegrityCheck::RsGxsIntegrityCheck(
         RsGeneralDataService* const dataService, RsGenExchange* genex,
-        RsSerialType& serializer, RsGixs* gixs ) :
-    mDs(dataService), mGenExchangeClient(genex), mSerializer(serializer),
+        RsSerialType& /*serializer*/, RsGixs* gixs ) :
+    mDs(dataService), mGenExchangeClient(genex),/* mSerializer(serializer),*/
```
Comment (G10h4ck) [inline libretroshare/src/gxs/rsgxsutil.cc]: Are you sure this won't break deep search? (it is not compiled by default but will set as default soon)

File: libretroshare/src/gxs/rsgxsutil.h [hunk 11]
```diff
@@ -214,7 +213,7 @@ class RsGxsIntegrityCheck : public RsSingleJobThread
 
 	RsGeneralDataService* const mDs;
 	RsGenExchange *mGenExchangeClient;
-	RsSerialType& mSerializer;
+	//RsSerialType& mSerializer;
```
Comment (G10h4ck) [inline libretroshare/src/gxs/rsgxsutil.h]: This breaks deep search, consider putting it and related changes inside and `#ifdef` instead

File: libretroshare/src/services/p3gxschannels.h [hunk 12]
```diff
@@ -24,11 +24,11 @@
 #define P3_GXSCHANNELS_SERVICE_HEADER
```
Comment (G10h4ck) [inline libretroshare/src/services/p3gxschannels.h:24]: Editing this file will probably give a bunch of merging conflicts with safer gxschannel API PR

File: libretroshare/src/services/p3gxscircles.cc [hunk 13]
```diff
@@ -963,9 +963,9 @@ bool p3GxsCircles::cache_request_load(const RsGxsCircleId &id)
 	int32_t age = 0;
 	if (RsTickEvent::prev_event_ago(CIRCLE_EVENT_CACHELOAD, age))
 	{
-		if (age < MIN_CIRCLE_LOAD_GAP)
+		if( (age < MIN_CIRCLE_LOAD_GAP) && (age > 0))
```
Comment (G10h4ck) [inline libretroshare/src/services/p3gxscircles.cc]: @csoler I think that if we get age < 0 at this point at least a warning must be printed

File: libretroshare/src/tcponudp/tcpstream.cc [hunk 14]
```diff
@@ -2648,7 +2667,7 @@ int	setupBinaryCheck(std::string fname)
 /* uses seq number to track position -> ensure no rollover */
 int	checkData(uint8 *data, int size, int idx)
 {
-	if (bc_fd <= 0)
+	if (bc_fd = 0)
```
Comment (G10h4ck) [inline libretroshare/src/tcponudp/tcpstream.cc]: This assignment here seems a very dangerous typo!!

File: libretroshare/src/turtle/p3turtle.cc [hunk 15]
```diff
@@ -393,7 +388,7 @@ void p3turtle::manageTunnels()
 
 			static const float grow_speed = 1.0f ;	// speed at which the time increases.
 
-			float tunnel_keeping_factor = (std::max(1.0f,(float)total_speed/(float)(50*1024)) - 1.0f)*grow_speed + 1.0f ;
+			float tunnel_keeping_factor = (std::max(1.0f,static_cast<float>(total_speed)/(50.0f*1024)) - 1.0f)*grow_speed + 1.0f ;
```
Comment (G10h4ck) [inline libretroshare/src/turtle/p3turtle.cc]: @csoler it's just a coincidence it has same value as `TUNNEL_REQUEST_PACKET_SIZE_APPROX` ?

File: libretroshare/src/turtle/p3turtle.cc [hunk 16]
```diff
@@ -2208,11 +2205,12 @@ static std::string printNumber(uint64_t num,bool hex=false)
 	if(hex)
 	{
 		char tmp[100] ;
+		uint64_t one = 1;
```
Comment (G10h4ck) [inline libretroshare/src/turtle/p3turtle.cc]: Compiler might figure out anyway but better to specify `constexpr` anyway

File: retroshare-gui/src/TorControl/TorManager.cpp [hunk 17]
```diff
@@ -150,7 +152,7 @@ void TorManager::setHiddenServiceDirectory(const QString &path)
 
 bool TorManager::setupHiddenService()
 {
-	if(d->hiddenService != NULL)
+	if(d->hiddenService != nullptr)
```
Comment (G10h4ck) [inline retroshare-gui/src/TorControl/TorManager.cpp]: Better `if(d->hiddenService)`

File: retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp [hunk 18]
```diff
@@ -141,15 +141,15 @@ class RsDownloadListModel : public QAbstractItemModel
...
 #endif
-		return mDownloads[entry].peers.size();
+		return static_cast<int>(mDownloads[entry].peers.size());
 	}
-	int columnCount(const QModelIndex &parent = QModelIndex()) const
+	int columnCount(const QModelIndex &/*parent*/ = QModelIndex()) const
 	{
 		return COLUMN_COUNT ;
 	}
 	bool hasChildren(const QModelIndex &parent = QModelIndex()) const
```
Comment (G10h4ck) [inline retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:150]: It is usually better to expose into header instead of that in the implementation that a function has default parameter value

File: retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp [hunk 19]
```diff
@@ -180,15 +180,15 @@ class RsDownloadListModel : public QAbstractItemModel
...
 		uint32_t entry = 0;
 		int source_id=0 ;
 
 		if(!parent_ref)	// top level. The entry is that of a transfer
 		{
-			void *ref = NULL ;
+			void *ref = nullptr ;
 
-			if(row >= (int)mDownloads.size() || !convertTabEntryToRefPointer(row,-1,ref))
+			if(row >= static_cast<int>(mDownloads.size()) || !convertTabEntryToRefPointer(static_cast<uint32_t>(row),-1,ref))
```
Comment (G10h4ck) [inline retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:191]: @csoler can you confirm this is ok? It reminds me of the 64/32 bit file node reference depending on architecture

File: retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp [hunk 20]
```diff
@@ -396,29 +428,28 @@ class RsDownloadListModel : public QAbstractItemModel
...
-					priority = 0;
+					priority = PRIORITY_NULL;
 				else
 					switch (fileInfo.priority)
 					{
 					case SPEED_LOW:     priority = PRIORITY_SLOWER; break;
 					case SPEED_NORMAL:  priority = PRIORITY_AVERAGE; break;
 					case SPEED_HIGH:    priority = PRIORITY_FASTER; break;
-					default:            priority = PRIORITY_AVERAGE; break;
+					//default:            priority = PRIORITY_AVERAGE; break;
```
Comment (G10h4ck) [inline retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:445]: this probably will introduce a warning of missing default case on some compilers

File: retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp [hunk 21]
```diff
@@ -1109,6 +1114,7 @@ void TransfersDialog::activatePage(TransfersDialog::Page page)
 													break ;
 		case TransfersDialog::RemoteSharedFilesTab: ui.tabWidget->setCurrentWidget(remoteSharedFiles) ;
 													break ;
+		case TransfersDialog::DownloadTab: assert(false);
```
Comment (G10h4ck) [inline retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp]: won't this just crash retroshare in that case? Better printing some information about errors/stacktrace

**Review by G10h4ck — CHANGES_REQUESTED** [review]
Next time avoid doing pull requests so huge, it take too much time reviewing them and keeping high level of attention it's hard! Also when you do warning fixing  try to do one pull request for king of warning, for example a pull request where you only change NULL into nullptr  and then another pull request where you static_cast some integers types, and another where you change the if to check directly the pointers without comparing are much easier to review and less error prone then a pull request with the three things all toghether

File: libretroshare/src/ft/ftserver.cc (hunk 6 above)
Comment (PhenomRetroShare) [reply libretroshare/src/ft/ftserver.cc]: No, just to see that we don't return object but if pointer is not null.

File: retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp (hunk 18 above)
Comment (PhenomRetroShare) [reply retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:150]: The class is defined here, no header. ;)

File: retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp (hunk 19 above)
Comment (PhenomRetroShare) [reply retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:191]: int row
static bool convertTabEntryToRefPointer(uint32_t entry,int source_id,void *& ref)

File: retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp (hunk 20 above)
Comment (PhenomRetroShare) [reply retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:445]: This is to remove once:
    /retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:414: warning:
    default label in switch which covers all enumeration values

Comment (G10h4ck) [reply retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:445]: ok!

File: libretroshare/src/ft/ftextralist.cc (hunk 4 above)
Comment (PhenomRetroShare) [reply libretroshare/src/ft/ftextralist.cc]: Done

File: libretroshare/src/ft/ftextralist.cc (hunk 5 above)
Comment (PhenomRetroShare) [reply libretroshare/src/ft/ftextralist.cc]: Done

File: libretroshare/src/gxs/rsgxsnetservice.cc (hunk 8 above)
Comment (PhenomRetroShare) [reply libretroshare/src/gxs/rsgxsnetservice.cc]: Done

File: libretroshare/src/gxs/rsgxsnetservice.cc (hunk 9 above)
Comment (PhenomRetroShare) [reply libretroshare/src/gxs/rsgxsnetservice.cc]: Done

File: libretroshare/src/gxs/rsgxsutil.cc (hunk 10 above)
Comment (PhenomRetroShare) [reply libretroshare/src/gxs/rsgxsutil.cc]: Done, I don't understand how I can make it possible. Sorry. ;)

File: libretroshare/src/gxs/rsgxsutil.h (hunk 11 above)
Comment (PhenomRetroShare) [reply libretroshare/src/gxs/rsgxsutil.h]: Done

File: libretroshare/src/services/p3gxscircles.cc (hunk 13 above)
Comment (PhenomRetroShare) [reply libretroshare/src/services/p3gxscircles.cc]: Done

File: libretroshare/src/tcponudp/tcpstream.cc (hunk 14 above)
Comment (PhenomRetroShare) [reply libretroshare/src/tcponudp/tcpstream.cc]: Shame on  me...

File: libretroshare/src/turtle/p3turtle.cc (hunk 15 above)
Comment (PhenomRetroShare) [reply libretroshare/src/turtle/p3turtle.cc]: I think it isn't.
Replaced

File: libretroshare/src/turtle/p3turtle.cc (hunk 16 above)
Comment (PhenomRetroShare) [reply libretroshare/src/turtle/p3turtle.cc]: Done

File: retroshare-gui/src/TorControl/TorManager.cpp (hunk 17 above)
Comment (PhenomRetroShare) [reply retroshare-gui/src/TorControl/TorManager.cpp]: Done

File: retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp (hunk 21 above)
Comment (PhenomRetroShare) [reply retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp]: Found better: Show DownloadTab ;)

File: libretroshare/src/ft/ftextralist.cc (hunk 5 above)
Comment (G10h4ck) [reply libretroshare/src/ft/ftextralist.cc]: It still appears as `if (fd == nullptr)` in the changes, is just GH that is outdated?

File: retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp (hunk 20 above)
Comment (csoler) [reply retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:445]: sorry but I really dont get it. The Default is needed here.I dont see why the compiler would complain. what if priority has another value than the possible values? We always need to make sure that kind of errors are handled.

File: libretroshare/src/tcponudp/tcpstream.cc [hunk 22]
```diff
@@ -2648,7 +2667,7 @@ int	setupBinaryCheck(std::string fname)
 /* uses seq number to track position -> ensure no rollover */
 int	checkData(uint8 *data, int size, int idx)
 {
-	if (bc_fd <= 0)
+	if (bc_fd == 0)
```
Comment (G10h4ck) [inline libretroshare/src/tcponudp/tcpstream.cc]: despite the misleading name `bc_fd` is a pointer so check it as a pointer and avoid arithmetic

`if(!bc_fd)`

**Review by G10h4ck — CHANGES_REQUESTED** [review]

File: libretroshare/src/gxs/rsgxsutil.cc (hunk 10 above)
Comment (G10h4ck) [reply libretroshare/src/gxs/rsgxsutil.cc]: It' because you are compiling without `CONFIG+=rs_deep_search`

File: libretroshare/src/tcponudp/tcpstream.cc (hunk 22 above)
Comment (PhenomRetroShare) [reply libretroshare/src/tcponudp/tcpstream.cc]: Done.
But new commits was waiting so they are added before you req.

File: retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp (hunk 20 above)
Comment (PhenomRetroShare) [reply retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:445]: @csoler you'll get a warning that not all enum are used.
Default value is set in initialization.
double priority = PRIORITY_AVERAGE;

File: libretroshare/src/gxs/rsgxsutil.cc (hunk 10 above)
Comment (PhenomRetroShare) [reply libretroshare/src/gxs/rsgxsutil.cc]: CppCheck checks all this case normally.
And I generally check were removed values are used by searching by text in all trunk. Perhaps I was tired. ;)

File: libretroshare/src/turtle/p3turtle.cc (hunk 15 above)
Comment (G10h4ck) [reply libretroshare/src/turtle/p3turtle.cc]: ping @csoler

File: libretroshare/src/ft/ftserver.cc [hunk 23]
```diff
@@ -1356,7 +1357,7 @@ bool ftServer::decryptItem(const RsTurtleGenericDataItem *encrypted_item,const R
 	decrypted_item = dynamic_cast<RsTurtleGenericTunnelItem*>(deserialise(data,&data_size)) ;
 	free(data);
 
-	return (decrypted_item != NULL);
```
Comment (csoler) [inline libretroshare/src/ft/ftserver.cc:1359]: This is the limit to what I can accept: the test returns a bool as an explicit calculation. If you return the object itself, it is implicitely converted to a bool, which is much less clear and prone to wrong interpretation (not to mention the syntax with () that are not needed). So I would definitely keep the if(decrypted_item != NULL) here.

File: libretroshare/src/ft/ftserver.cc [hunk 24]
```diff
@@ -1481,16 +1482,18 @@ TurtleSearchRequestId ftServer::turtleSearch(const RsRegularExpression::Lineariz
     return mTurtleRouter->turtleSearch(expr) ;
 }
 
-#warning we should do this here, but for now it is done by turtle router.
+#ifndef RS_NO_WARN_CPP
+#pragma message("csoler 2018-06-10: We should do this here, but for now it is done by turtle router.")
+#endif
 //   // Dont delete the item. The client (p3turtle) is doing it after calling this.
 //   //
-//   void ftServer::receiveSearchResult(RsTurtleSearchResultItem *item)
```
Comment (csoler) [inline libretroshare/src/ft/ftserver.cc:1487]: How can the compiler generate warnings in a section that is a comment ???

File: libretroshare/src/ft/ftserver.cc (hunk 7 above)
Comment (csoler) [reply libretroshare/src/ft/ftserver.cc:1559]: Definitely a chain reaction ;-) I would keep it as it is.

File: libretroshare/src/ft/ftserver.cc [hunk 25]
```diff
@@ -1724,9 +1727,9 @@ int ftServer::handleIncoming()
 	// now File Input.
 	int nhandled = 0 ;
 
-	RsItem *item = NULL ;
+	RsItem *item = nullptr ;
 
-	while(NULL != (item = recvItem()))
+	while( (item = recvItem()) )
```
Comment (csoler) [inline libretroshare/src/ft/ftserver.cc]: Well, that is also reported by my compiler as a possible mistake, and I fully agree with him. Please keep the NULL (or change it to nullptr if you need)

File: libretroshare/src/ft/ftserver.cc [hunk 26]
```diff
@@ -1827,7 +1830,7 @@ int ftServer::handleIncoming()
  **********************************
  *********************************/
 
-void ftServer::receiveSearchResult(RsTurtleFTSearchResultItem *item)
```
Comment (csoler) [inline libretroshare/src/ft/ftserver.cc:1830]: why this change of names? This function overloads the method in TurtleClientService. So it should not be changed! If you do, FT will just not work anymore.

File: libretroshare/src/gxs/rsgxsnetservice.cc [hunk 27]
```diff
@@ -714,8 +714,8 @@ void RsGxsNetService::syncWithPeers()
 
             // The last post will be set to TS 0 if the req delay is 0, which means "Indefinitly"
 
-            if(req_delay > 0)
-				msg->createdSinceTS = std::max(0,(int)time(NULL) - req_delay);
```
Comment (csoler) [inline libretroshare/src/gxs/rsgxsnetservice.cc:718]: (int)something  *is* a static cast. So why convert all the casts in the source into explicit static<cast>(int)? This makes the whole code much less readable.

File: libretroshare/src/gxs/rsgxsnetservice.cc [hunk 28]
```diff
@@ -747,12 +747,12 @@ void RsGxsNetService::generic_sendItem(RsNxsItem *si)
 
 	RsGxsGroupId tmp_grpId;
 
-	if(mAllowDistSync && mGxsNetTunnel != NULL && mGxsNetTunnel->isDistantPeer( static_cast<RsGxsNetTunnelVirtualPeerId>(si->PeerId()),tmp_grpId))
```
Comment (csoler) [inline libretroshare/src/gxs/rsgxsnetservice.cc:750]: when I read this, it reminds me that mGxsNetTunnel is a pointer. That is why it is compared to a pointer. Implicily casting to bool for testing is far worse IMHO.

File: libretroshare/src/gxs/rsgxsnetservice.cc [hunk 29]
```diff
@@ -877,7 +879,7 @@ void RsGxsNetService::syncGrpStatistics()
 			for(uint32_t i=0;i<n;++i)
 				++rit ;
 
-			for(uint32_t i=0;i<std::min(rec.suppliers.ids.size(),(size_t)GROUP_STATS_UPDATE_NB_PEERS);++i)
+			for(size_t i=0;i<std::min(rec.suppliers.ids.size(),GROUP_STATS_UPDATE_NB_PEERS);++i)
```
Comment (csoler) [inline libretroshare/src/gxs/rsgxsnetservice.cc:882]: I prefer the previous version: .size() is a size_t, so the int GROUP_STATS... should be cast to size_t. So in summary: keep GROUP_STATS_... to be an int. Keep the size_t in the for()

File: libretroshare/src/gxs/rsgxsnetservice.cc [hunk 30]
```diff
@@ -960,7 +962,7 @@ void RsGxsNetService::handleRecvSyncGrpStatistics(RsNxsSyncGrpStatsItem *grs)
 
 	    RsNxsSyncGrpStatsItem *grs_resp = new RsNxsSyncGrpStatsItem(mServType) ;
 	    grs_resp->request_type = RsNxsSyncGrpStatsItem::GROUP_INFO_TYPE_RESPONSE ;
-	    grs_resp->number_of_posts = vec.size();
+	    grs_resp->number_of_posts = static_cast<uint32_t>(vec.size());
```
Comment (csoler) [inline libretroshare/src/gxs/rsgxsnetservice.cc:965]: is that really necessary?

**Comment by csoler** [comment]
ok. I've read 20% of this PR. As much as I appreciate contributions to clean and improve the code, I think that they are generic problems here:
- systematically using explicit static_cast<>() instead of (type)something makes the code much less readable. 
- using implicit cast of pointers to bool in comparisons, and especially in boolean return of functions makes the code much less readable
- fixing "warnings" in parts of the code that are commented out does not make sense to me.
- NULL is valid but outdated. So we use nullptr in new code, but changing all NULLs into nullptr looks like a significant waste of time IMHO.

I really do not understand why you spend so much time making these changes. Their non critical nature does not compare favorably with the effort that they need devs proofread them, and the very high risk of subtle errors they can bring in the code  (not because its you who do them, but rather because of their nature and repetitive occurrence).

File: libretroshare/src/ft/ftserver.cc (hunk 23 above)
Comment (PhenomRetroShare) [reply libretroshare/src/ft/ftserver.cc:1359]: I'm OK with that.

File: libretroshare/src/ft/ftserver.cc (hunk 24 above)
Comment (PhenomRetroShare) [reply libretroshare/src/ft/ftserver.cc:1487]: This code i commented and could be uncommented. That's why I change it too.

File: libretroshare/src/ft/ftserver.cc (hunk 25 above)
Comment (PhenomRetroShare) [reply libretroshare/src/ft/ftserver.cc]: OK

File: libretroshare/src/ft/ftserver.cc (hunk 26 above)
Comment (PhenomRetroShare) [reply libretroshare/src/ft/ftserver.cc:1830]: Looks at warning:
/libretroshare/src/ft/ftserver.h:101: warning:
'ftServer::receiveSearchResult' hides overloaded virtual function [-
Woverloaded-virtual]
/libretroshare/src/turtle/turtleclientservice.h:122: hidden overloaded
virtual function 'RsTurtleClientService::receiveSearchResult' declared
here: different number of parameters (3 vs 1)

File: libretroshare/src/gxs/rsgxsnetservice.cc (hunk 27 above)
Comment (PhenomRetroShare) [reply libretroshare/src/gxs/rsgxsnetservice.cc:718]: /libretroshare/src/gxs/rsgxsnetservice.cc:718: warning: use of old-style
cast
/libretroshare/src/gxs/rsgxsnetservice.cc:718: warning: implicit
conversion changes signedness: 'const int' to 'uint32_t' (aka 'unsigned
int')

File: libretroshare/src/gxs/rsgxsnetservice.cc (hunk 28 above)
Comment (PhenomRetroShare) [reply libretroshare/src/gxs/rsgxsnetservice.cc:750]: \<Joke>Could be rename to mpGxsNetTunnel and mbAllowDistSync.\</joke> ;)
And we use it just after.
@G10h4ck Need to be clarified in Coding style.

File: libretroshare/src/gxs/rsgxsnetservice.cc (hunk 29 above)
Comment (PhenomRetroShare) [reply libretroshare/src/gxs/rsgxsnetservice.cc:882]: GROUP_STATS_UPDATE_NB_PEERS is only used here.

File: libretroshare/src/gxs/rsgxsnetservice.cc (hunk 30 above)
Comment (PhenomRetroShare) [reply libretroshare/src/gxs/rsgxsnetservice.cc:965]: You're right, number_of_posts should be size_t. ;)
/libretroshare/src/gxs/rsgxsnetservice.cc:963: warning: implicit
conversion loses integer precision: 'std::vector::size_type' (aka
'unsigned long') to 'uint32_t' (aka 'unsigned int')
size_t depends on system.

File: libretroshare/src/ft/ftserver.cc (hunk 26 above)
Comment (PhenomRetroShare) [reply libretroshare/src/ft/ftserver.cc:1830]: This is a relevant warning...

**Comment by PhenomRetroShare** [comment]
>     * systematically using explicit static_cast<>() instead of (type)something makes the code much less readable.
warning: use of old-style cast

>     * using implicit cast of pointers to bool in comparisons, and especially in boolean return of functions makes the code much less readable
I'm agreed with that.

>     * NULL is valid but outdated. So we use nullptr in new code, but changing all NULLs into nullptr looks like a significant waste of time IMHO.
Only changed in modified file.

> I really do not understand why you spend so much time making these changes.

Only to spent less time to see if what I'll code add new warnings.
And to see relevant ones clearly.
More than 60000 in trunk.
scrollbar
For only one file...
That's make a long time we don't make a coverity scan...

File: libretroshare/src/ft/ftserver.cc (hunk 26 above)
Comment (csoler) [reply libretroshare/src/ft/ftserver.cc:1830]: yes I agree. The very weird thing here is that I'm pretty sure to have fixed this already, but apparently I did not. Gio told me the exact same thing weeks ago, and I thought I fixed it.
So: yes, change the same to ftServerReceiveSearchResults() and call this function in p3turtle.cc:1223  instead of             client->receiveSearchResult(ftsr);

**Comment by G10h4ck** [comment]
I think cleaning up the code make sense, but it must be done in little PR so we can manage them, this PR is too huge to keep track of reviewing... @PhenomRetroShare to avoid your work to get lost, would you like to split this in little PRs that we can more reasonably review and agree on?

**Comment by csoler** [comment]
If you do so, please:
- don't change all NULL into nullptr. It's a #define, so there's no point in doing that
- don't implicitly cast pointers to bool in return of functions. 
- don't replace c-style static casts into c++ static_cast<>. It's the same, and it makes the code much harder to read.

**Comment by G10h4ck** [comment]
> don't replace c-style static casts into c++ static_cast<>. It's the same, and it makes the code much harder to read.

This is problematic because C style cast may cause any of reinterpret, const, dynamic, or static cast and even some of them together unintended, in addiction to that not only cpp-check but also compilers reports that as problem and getting many of those warnings make it difficult to find important ones, so we need to use explicit casting in new code and think as a way to port the old code to the C++ style cast without making it unreadable or the PR impossible to review

**Comment by PhenomRetroShare** [comment]
It's possible to cherry-pick commit by commit...
I cannot make a PR for each modified files.
I don't see a better manner to do it.
And we have to do it before running a coverity scan.
Once done, no need to do it later... or at least they will be very little.

**Comment by G10h4ck** [comment]
Yeah no need of a PR for each file, but it must be reasonable to review the code and maximum ten minutes for this kind (fix compiler warning, style improvements...) of PR

**Comment by PhenomRetroShare** [comment]
Cherry-pick by commits so...

File: libretroshare/src/ft/ftserver.cc (hunk 26 above)
Comment (PhenomRetroShare) [reply libretroshare/src/ft/ftserver.cc:1830]: OK, but I test if it is a ftServer before.
